superset-frontend/src/views/CRUD/data/database/DatabaseModal/ModalHeader.tsx

```diff
@@ -142,19 +154,23 @@ const ModalHeader = ({
     </StyledFormHeader>
   );
 
+  const importDbHeader = (
+    <StyledStickyHeader>
+      <StyledFormHeader>
+        <p className="helper-top"> STEP 2 OF 2 </p>
+        <h4>Enter the required {dbModel.name} credentials</h4>
+        <p className="helper-bottom">{fileCheck ? file[0].name : ''}</p>
+      </StyledFormHeader>
+    </StyledStickyHeader>
+  );
+
+  if (fileCheck) return importDbHeader;
   if (isLoading) return <></>;
-  if (isEditMode) {
-    return isEditHeader;
-  }
-  if (useSqlAlchemyForm) {
-    return useSqlAlchemyFormHeader;
-  }
-  if (hasConnectedDb && !editNewDb) {
-    return hasConnectedDbHeader;
-  }
-  if (db || editNewDb) {
-    return hasDbHeader;
-  }
+  if (isEditMode) return isEditHeader;
+  if (useSqlAlchemyForm) return useSqlAlchemyFormHeader;
+  if (hasConnectedDb && !editNewDb) return hasConnectedDbHeader;
+  if (db || editNewDb) return hasDbHeader;
```

Comment:
We could also use a switch here:

```js
switch (true) {
  fileCheck: return importDbHeader;
  isLoading: return <></>;
  isEditMode: return isEditHeader;
  useSqlAlchemyForm: return useSqlAlchemyFormHeader;
  hasConnectedDb && !editNewDb: return hasConnectedDbHeader;
  db || editNewDb: return hasDbHeader;
  default: noDbHeader;
}
```

Reply:
I wrote it out like this (at the linter's request):
```switch (true) {
    case fileCheck:
      return importDbHeader;
    case isLoading:
      return <></>;
    case isEditMode:
      return isEditHeader;
    case useSqlAlchemyForm:
      return useSqlAlchemyFormHeader;
    case hasConnectedDb && !editNewDb:
      return hasConnectedDbHeader;
    case db || editNewDb:
      return hasDbHeader;
    default:
      return noDbHeader;
  }
```
And it will show all headers except `hasDbHeader` - is it possible I structured this incorrectly? I don't see why it wouldn't work but it does not 😅

Reply:
Ah, I had forgotten the `case`. I wouldn't stress about, I'm not even sure if we can use `return` inside a `switch`, they're weird beasts. :-P

Reply:
They are weird, but cool when you can use them! I'll leave this as is for now in this case 😁